mantle/platform/qemu.go

```diff
@@ -115,14 +122,73 @@ func (inst *QemuInstance) SSHAddress() (string, error) {
 	return "", fmt.Errorf("didn't find an address")
 }
 
+// Wait for the qemu process to exit
 func (inst *QemuInstance) Wait() error {
 	return inst.qemu.Wait()
 }
 
+// WaitIgnitionError will only return if the instance
+// failed inside the initramfs.  The resulting string will
+// be a newline-delimited stream of JSON strings, as returned
+// by `journalctl -o json`.
+func (inst *QemuInstance) WaitIgnitionError() (string, error) {
+	b := bufio.NewReaderSize(inst.journalPipe, 64768)
+	var r strings.Builder
+	iscorrupted := false
+	_, err := b.Peek(1)
+	if err != nil {
+		if err == io.EOF {
+			return "", nil
+		}
+		return "", errors.Wrapf(err, "Reading from journal")
+	}
+	for {
+		line, prefix, err := b.ReadLine()
+		if err != nil {
+			return r.String(), errors.Wrapf(err, "Reading from journal channel")
+		}
+		if prefix {
+			iscorrupted = true
+		}
+		if len(line) == 0 || string(line) == "{}" {
+			break
+		}
+		r.Write(line)
+	}
+	if iscorrupted {
+		return r.String(), fmt.Errorf("journal was truncated due to overly long line")
+	}
+	return r.String(), nil
+}
+
+// WaitAll wraps the process exit as well as WaitIgnitionError,
+// returning an error if either fail.
+func (inst *QemuInstance) WaitAll() error {
```

Comment:
Hmm, do we actually need this new interface? Can we have it part of `Wait()` directly and have that return `ErrInitramfsEmergency` if Ignition failed? That way we automatically get this check everywhere we use and will use `Wait()`.

Reply:
Funny you ask.  I actually did it that way first.  And then, you know, I went to go test out a FCOS build failing in the initramfs (which btw if you want to conveniently test, use e.g.
`cosa run --kargs ignition.config.url=blah://`) and qemu started, printed an error about the initramfs failure and then exited...so I couldn't debug it.

After that I made it a separate interface, because for that reason we don't always want to die if there's a failure in the initramfs.   (Now, theoretically the caller could distinguish and explicitly call `Wait()` again but that's racy because there's already another goroutine doing that...)